src/components/views/joinGame/JoinGame.tsx

```diff
@@ -0,0 +1,61 @@
+import BaseContainer from "../../ui/BaseContainer"
+import React, { useContext, useState } from "react"
+import Button from "@mui/material/Button"
+import SendIcon from "@mui/icons-material/Send"
+
+import { useGames } from "../../../hooks/api/useGames"
+import { Paths } from "../../routing/routers/Paths"
+import { useNavigate, useParams } from "react-router-dom"
+import { useApi } from "../../../hooks/api/useApi"
+import { appContext } from "../../../AppContext"
+
+const JoinGame = () => {
+  const { joinGame } = useGames()
+  const navigate = useNavigate()
+  const gameId = useParams()
+
+  const [loading, setLoading] = useState(false)
+  const { wrapApiCall, gameEntityController } = useApi()
+  const gameStore = useContext(appContext).playerStore
+
+  const findGame = async (gameId) => {
+    setLoading(true)
+    const foundGame = await wrapApiCall(
+      gameEntityController.getItemResourceGameGet("{gameId}")
+    ).finally(() => setLoading(false))
+    gameStore.setMe(foundGame)
```

Comment:
The hope was that we don't need to change the store in the components, but can wrap api calls and store changes in the hooks and expose it as one operation.